src/main/scala/temple/detail/LanguageDetail.scala

```diff
@@ -0,0 +1,15 @@
+package temple.detail
+
+/** Maintains any extra required information the user did not provide in the Templefile */
+sealed trait LanguageDetail
```

Comment:
Must say I’m a little confused as to how this trait is going to be used

Reply:
The idea is that it gets passed with the Templefile into the ProjectBuilder, and can then get used to gain the extra detail needed

Reply:
I meant more that it seems orthogonal to the directions in which classes should go: if you call build, you get a LanguageDetail, but there’s no way to do anything with a LanguageDetail by itself other than pattern matching, when that pattern matching is parallel to the language itself

Reply:
I’m too tired to think of a solution now, but it seems like a bit of a strange use of the type system

Reply:
I’ll accept this for now, but may suggest something more when the PR using it comes in